WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,606 @@
+                if (isEmpty) {
+                    String message = getString(R.string.stock_media_picker_initial_empty_text);
+                    String subMessage = getString(R.string.stock_media_picker_initial_empty_subtext);
+                    String link = "<a href='https://pexels.com/'>Pexels</a>";
+                    String html = message
+                                  + "<br /><br />"
+                                  + "<small>" + String.format(subMessage, link) + "</small>";
+                    txtEmpty.setMovementMethod(WPLinkMovementMethod.getInstance());
+                    txtEmpty.setText(Html.fromHtml(html));
+                } else {
+                    txtEmpty.setText(R.string.stock_media_picker_empty_results);
+                }
+            }
+        }
+    }
+
+    private void showProgress(boolean show) {
+        if (!isFinishing()) {
+            findViewById(R.id.progress).setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void submitSearch(@Nullable final String query, boolean delayed) {
+        mSearchQuery = query;
+
+        if (query == null || query.length() < MIN_SEARCH_QUERY_SIZE) {
+            mAdapter.clear();
+            showEmptyView(true);
+            return;
+        }
+
+        if (delayed) {
+            mHandler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    if (StringUtils.equals(query, mSearchQuery)) {
+                        submitSearch(query, false);
+                    }
+                }
+            }, 500);
+        } else {
+            fetchStockMedia(query, 1);
+        }
+    }
+
+    private void fetchStockMedia(@Nullable String searchQuery, int page) {
+        if (mIsFetching || !NetworkUtils.checkConnection(this)) {
+            return;
+        }
+
+        if (TextUtils.isEmpty(searchQuery)) {
+            mAdapter.clear();
+            return;
+        }
+
+        if (page == 1) {
+            mAdapter.clear();
+        }
+
+        showProgress(true);
+        mIsFetching = true;
+        showEmptyView(false);
+
+        mSearchQuery = searchQuery;
+        AppLog.d(AppLog.T.MEDIA, "Fetching stock media page " + page);
+
+        StockMediaStore.FetchStockMediaListPayload payload =
+                new StockMediaStore.FetchStockMediaListPayload(searchQuery, page);
+        mDispatcher.dispatch(StockMediaActionBuilder.newFetchStockMediaAction(payload));
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onStockMediaListFetched(StockMediaStore.OnStockMediaListFetched event) {
+        // make sure these results are for the same query
+        if (mSearchQuery == null || !mSearchQuery.equals(event.searchTerm)) {
+            return;
+        }
+
+        mIsFetching = false;
+        showProgress(false);
+
+        if (event.isError()) {
```

Comment:
When an error occurs, do we need to show the empty view with the message? Or maybe show some kind of toast? It looks like it's silently failing right now. Please ignore the comment if you were planning to handle this later.

Reply:
I'll add a generic error toast here, but keep in mind this endpoint very rarely will be an error since it returns an HTTP 200 with an empty media list when a timeout or other server issue arises.